apps/prairielearn/src/lib/score-upload.ts

```diff
@@ -394,29 +394,36 @@ async function updateAssessmentInstanceFromJson(
   assessment_id: string,
   authn_user_id: string,
 ) {
-  let query, id;
-  if (_.has(json, 'uid')) {
-    query = sql.select_assessment_instance_uid;
-    id = json.uid;
-  } else if (_.has(json, 'group_name')) {
-    query = sql.select_assessment_instance_group;
-    id = json.group_name;
-  } else {
-    throw new Error('"uid" or "group_name" not found');
-  }
   if (!_.has(json, 'instance')) throw new Error('"instance" not found');
-
   await sqldb.runInTransactionAsync(async () => {
-    const assessment_instance_id = await sqldb.queryOptionalRow(
-      query,
-      {
-        assessment_id,
-        uid: json.uid,
-        group_name: json.group_name,
-        instance_number: json.instance,
-      },
-      IdSchema,
-    );
+    let assessment_instance_id: string | null;
+    let id: string | null;
+    if (_.has(json, 'uid')) {
+      id = json.uid;
+      assessment_instance_id = await sqldb.queryOptionalRow(
+        sql.select_assessment_instance_uid,
```

Comment:
Nit: I've started moving things like this into a separate function, as I think it makes control flow easier to follow (and it's easier for TypeScript to help you out):

```ts
function getAssessmentInstanceId(json, ...) {
  if ('uid' in json) {
    return {
      id: json.uid,
      assessment_instance_id: await queryOptionalRow(...)
    };
  }

  if ('group_name' in json) {
    // ...
  }

  throw new Error('...')
}

// Later...
const { assessment_instance_id, id } = await getAssessmentInstanceId();
```

I'm not blocking on this comment.

Reply:
I gave that a shot in e61d92f

Reply:
LGTM!